### PR TITLE
Mechs now can't be spam repaired

### PR DESCRIPTION
**Pull request by farie82**
**What does this PR do:**
Mechs now need time to be repaired and you can only repair it once per ... well repair. No spam repairing anymore
Every repair costs 1.5 seconds now.

fixes: #10041

**Changelog:**
:cl: Farie82
tweak: Repairing a mech now says you're doing it to the others
balance: Mechs now take time to repair. No more spam repairing
/:cl:

### CODE REVIEW

**Comment by Birdtalon** [comment]
A pretty big nerf to mechs imo.

**Comment by farie82** [comment]
If needed I can boost the amount it heals. Or just lower the time it takes to repair

**Comment by variableundefined** [comment]
Will need head approval for this one.

**Comment by Necaladun** [comment]
I seem to remember having to click a LOT to repair it, so might wanna adjust amount repaired/fuel used/etc. But I like this kinda stuff in theory - having an auto/rapidclicker shouldn't be an advantage.

**Comment by variableundefined** [comment]
@farie82 We've decided that increasing the fix number to 20 would be appropriate. 

No other requested changes here, just need to wait 24 hours.

**Comment by farie82** [comment]
@variableundefined Copy that. Done so

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -778,15 +780,22 @@
...
 			if (WT.remove_fuel(0,user))
-				if (internal_damage & MECHA_INT_TANK_BREACH)
-					clearInternalDamage(MECHA_INT_TANK_BREACH)
-					to_chat(user, "<span class='notice'>You repair the damaged gas tank.</span>")
+				if(!(user in repairers)) // Can only repair it once at a time
+					repairers += user
+					user.visible_message("<span class='notice'>[user] starts repairing some damage to [name].</span>", "<span class='notice'>You start repairing some damage to [name]</span>")
+					playsound(user, WT.usesound, 50, 1)
+					if(do_after(user, 15 * WT.toolspeed, target = src)) // Takes 1.5 seconds to repair
+						if (internal_damage & MECHA_INT_TANK_BREACH)
```
Comment (variableundefined) [inline code/game/mecha/mecha.dm]: Spacing.

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -778,15 +780,22 @@
...
+							clearInternalDamage(MECHA_INT_TANK_BREACH)
+							user.visible_message("<span class='notice'>[user] repairs the damaged gas tank.</span>", "<span class='notice'>You repair the damaged gas tank.</span>")
+						else
+							user.visible_message("<span class='notice'>[user] repairs some damage to [name].</span>", "<span class='notice'>You repair some damage to [name]</span>")
```
Comment (variableundefined) [inline code/game/mecha/mecha.dm]: Missing punctuation

File: code/game/mecha/mecha.dm [hunk 3]
```diff
@@ -778,15 +780,22 @@
...
+							health += min(20, initial(health)-health)
```
Comment (variableundefined) [inline code/game/mecha/mecha.dm]: Spacing between operator & operatee please aka initial(health) - health

**Comment by variableundefined** [comment]
Hold up on fixing this PR - A better solution might be coming so I would not like you to waste your efforts on this.

**Comment by tigercat2000** [comment]
Alright, I came up with a solution, I'll let you take it and merge this PR because I'm getting irritated with my larger mech PR and I don't mind just fixing the conflicts when I'm less frustrated at this point. 
Basically, do this:
code/__HELPERS/mobs.dm#L375 (basically shove it under do_after):
```dm
#define DOAFTERONCE_MAGIC "Magic~~"
GLOBAL_LIST_INIT(do_after_once_tracker, list())
/proc/do_after_once(mob/user, delay, needhand = 1, atom/target = null, progress = 1)
	if(!user || !target)
		return

	var/cache_key = "[user.UID()][target.UID()]"
	if(GLOB.do_after_once_tracker[cache_key])
		GLOB.do_after_once_tracker[cache_key] = DOAFTERONCE_MAGIC
		to_chat(user, "<span class='warning'>Attempt cancelled.</span>")
		return FALSE
	GLOB.do_after_once_tracker[cache_key] = TRUE
	. = do_after(user, delay, needhand, target, progress, extra_checks = CALLBACK(GLOBAL_PROC, .proc/do_after_once_checks, cache_key))
	GLOB.do_after_once_tracker[cache_key] = FALSE

/proc/do_after_once_checks(cache_key)
	if(GLOB.do_after_once_tracker[cache_key] && GLOB.do_after_once_tracker[cache_key] == DOAFTERONCE_MAGIC)
		GLOB.do_after_once_tracker[cache_key] = FALSE
		return FALSE
	return TRUE
```
code/game/mecha/mecha.dm#L782
```dm
			if(WT.remove_fuel(0, user))
				if(do_after_once(user, 15 * WT.toolspeed, target = src))
					if(internal_damage & MECHA_INT_TANK_BREACH)
						clearInternalDamage(MECHA_INT_TANK_BREACH)
						to_chat(user, "<span class='notice'>You repair the damaged gas tank.</span>")
					else
						user.visible_message("<span class='notice'>[user] repairs some damage to [name].</span>")
						health += min(10, initial(health)-health)
```

**Pull request merged**